### PR TITLE
add type validation to parent_env handling

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <maxamillion@fedoraproject.org>

### CODE REVIEW

File: dockerfile_parse/parser.py [hunk 1]
```diff
@@ -92,9 +92,12 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
 
         self.env_replace = env_replace
 
-        if parent_env:
+        if type(parent_env) is dict:
```
Comment (twaugh) [inline dockerfile_parse/parser.py]: Could use `isinstance ()`.

Comment (maxamillion) [inline dockerfile_parse/parser.py]: 👍

File: dockerfile_parse/parser.py [hunk 2]
```diff
@@ -92,9 +92,12 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
 
         self.env_replace = env_replace
 
-        if parent_env:
+        if isinstance(parent_env, dict):
             logger.debug("Setting inherited parent image ENV vars: %s", parent_env)
             self.parent_env = parent_env
+        elif parent_env != None:
+            logger.debug("Invalid parent image ENV provided: %s", parent_env)
```
Comment (lcarva) [inline dockerfile_parse/parser.py]: Don't we want this to raise an exception? It seems unnatural to just bury the problem.

Comment (twaugh) [inline dockerfile_parse/parser.py]: If we want to do that, maybe we should just `assert isinstance(parent_env, dict)`.

Comment (maxamillion) [inline dockerfile_parse/parser.py]: +1

**Pull request merged**